lib/health/exercise.ex

```diff
@@ -0,0 +1,30 @@
+defmodule Health.Exercise do
+  @moduledoc """
+  An Exercise
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "exercises" do
+    field :name, :string
+    field :description, :string
+    field :difficulty, :integer
+
+    timestamps()
+  end
+
+  @doc false
+  @spec changeset(struct(), map()) :: %Ecto.Changeset{}
+  def changeset(exercise, attrs) do
+    exercise
+    |> cast(attrs, [:name, :description, :difficulty])
+    |> validate_required([:name, :description, :difficulty])
+    |> validate_number(:difficulty, greater_than: 0, less_than: 11)
+  end
+
+  @spec all :: [%Health.Exercise{}]
+  def all do
+    Health.Exercise |> Health.Repo.all
+  end
```

Comment:
Not sure I like this pattern. It's kind of ActiveRecord-ey...maybe I should just do the `Health.Repo.all` in the controller anyway

Reply:
Nope, that would be super un-phoenixy! By having the the interface in exercise.ex you have a single place to change out your data sources.

Reply:
It could probably use some more parameters like sort_by name to justify its existence :)

Reply:
yea it'll get there :)